### ISSUE TITLE
Deduplicate change to store via deep equality

### ISSUE THREAD

**Issue by flux627** [effector/effector#438]
I'm playing with the basic number counter example, and was pleased to find out that Effector seems to at least use shallow equality to determine if updates to a store should be reported:

```typescript
import { createEvent, createStore } from 'effector'

const add = createEvent<number>()
const sub = createEvent<number>()
const reset = createEvent()

const counter = createStore(0)
  .on(add, (count, num) => count + num)
  .on(sub, (count, num) => count - num)
  .reset(reset)

counter.watch(n => console.log('counter: ', n))

add(1)
sub(1)
sub(0)
reset()
```
This outputs only the changes to `counter`, filtering out events that create no change:
```
counter:  0
counter:  1
counter:  0
```

However, if we were to instead use an object, shallow equality fails and we no longer have these filtered:
```typescript
import { createEvent, createStore } from 'effector'

const add = createEvent<number>()
const sub = createEvent<number>()
const reset = createEvent()

const counter = createStore({ total: 0 })
  .on(add, (state, num) => ({ total: state.total + num }))
  .on(sub, (state, num) => ({ total: state.total - num }))
  .reset(reset)

counter.watch(n => console.log('counter: ', n))

add(1)
sub(1)
sub(0)

reset()
```
This outputs:
```
counter:  { total: 0 }
counter:  { total: 1 }
counter:  { total: 0 }
counter:  { total: 0 }
counter:  { total: 0 }
```

Is there a recommended, performant way to filter out changes via deep equality to the previous state?

**Comment by doasync** [comment]
There is a discussion related to your question. Check this answer and the link there:
https://github.com/effector/effector/discussions/428#discussioncomment-240507

**Comment by flux627** [comment]
Great, thank you! It seems that this feature isn't in the latest release yet. Though, good to know that it will be available soon.

**Comment by zerobias** [comment]
`updateFilter` released in [`effector 21.8.0`](https://github.com/effector/effector/releases/tag/effector%4021.8.0)
